core/src/main/java/org/hibernate/ogm/dialect/spi/TransactionContext.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.dialect.spi;
+
+import org.hibernate.resource.transaction.TransactionCoordinator.TransactionDriver;
+
+/**
+ * @author Davide D'Alto
+ */
+public interface TransactionContext {
+
+	TransactionDriver getTransactionDriver();
```

Comment:
@emmanuelbernard How do you feel about exposing `TransactionDriver` here? It's the right "scope" and it works, only I am wondering whether we should go for something abstract.

I.e. should we define a callback which allows to populate the context with the "business" data that the grid dialect operations are actually interested in (such as the TX id in Neo's case). Then `TransactionContext` would have a generic API a la `get(TransactionId.class)` or `get(Long.class)`, isolating grid dialects from the TX driver contract.

It's not strictly needed, but it seems a bit more extensible than fetching the state from the plain TX driver. Also it'd allow us to feed in generic TX state from within core down the road and dialects could access it (no idea whether we'll need it, but it'd be nice to have that option).

WDYT?